### ISSUE TITLE
[13.0] Include Exhausted Products is gone from Inventory Adjustment

### ISSUE THREAD

**Issue by andrenaught** [odoo/odoo#39398]
Impacted versions: 13.0

Steps to reproduce: Inventory -> Inventory Adjustment

Current behavior: The "Include Exhausted Products" check button is essentially gone. Currently there's no way to auto include all products unless their quantity on hand is above 0. The common use case here is doing initial stock, when you just add a bunch of products and then update their quantity all at once.

Expected behavior: There should be an option to include all products even if their quantity on hand is 0, like in odoo 12.0. Having the Product Category & Lot # option back would be cool too (less clicks) but I understand it's probably not necessary with the new product searching wizard.

**Comment by pedrobaeza** [comment]
@amoyaux what is the reasoning behind removing this option?

**Comment by m44rtin** [comment]
Any news?

We are considering moving to Odoo13 from an ancient warehouse management system but are hesitating since we cannot seem to make inventory lists by location that covers also products with zero quantity.
.... or am I missing something?

**Comment by amoyaux** [comment]
The issue with the exhausted product, is when you have multi location. In this case, if you have for example 100locations and 100 products it will generate immediately 100000 inventory lines. For the moment we decided to remove the option. However it's always possible to add missing lines manually or do an action that generate lines for you (it could be an idea since we can not reintroduce the feature as before in stable since we removed fields). Anyway we are aware of the issue and we are thinking about a solution.

**Comment by m44rtin** [comment]
> The issue with the exhausted product, is when you have multi location. In this case, if you have for example 100locations and 100 products it will generate immediately 100000 inventory lines. For the moment we decided to remove the option. However it's always possible to add missing lines manually or do an action that generate lines for you (it could be an idea since we can not reintroduce the feature as before in stable since we removed fields). Anyway we are aware of the issue and we are thinking about a solution.

Thanks for replying.

I created my inventory lists through the external API using the `stock.quant` table.
Perhaps this is an idea for you (with reservation for being a newbie on odoo).
If a product has an entry for the location, it should be included in the inventory list for that location.
In adition I also used the `stock.putaway.rule` for received products as (in my case) this determines the default location for my products.

In short here is how I approached making `stock.inventory.lines` for all my products:

1. Identify products with location and quantity from `stock.quant` (with some filtering)
2. Identify products with only location from `stock.putaway.rule`
3. Rest of products were given `/WH/Stock` and location and quant=0

This gave me inventory lines for my 5000+ products in my 1500+ locations.

**Comment by sle-odoo** [comment]
brought back at b5e24e9760a0d6a53730bebfc4bd76fadf971b3a
